tools/src/linter/SchemaVisitingValidator.ts

```diff
@@ -73,27 +72,23 @@ export default class SchemaVisitingValidator {
     }
 
     if (schema.type === 'number') {
-      if (schema.format === undefined || SCHEMA_NUMBER_FORMATS.includes(schema.format)) {
-        return
+      if (schema.format == null || !schema.format) {
+        errors.push(ctx.error(`Schema of type 'number' must specify a valid format. Allowed formats: ${SCHEMA_NUMBER_FORMATS.join(', ')}`));
+        return;
       }
-
-      if (SCHEMA_INTEGER_FORMATS.includes(schema.format)) {
-        errors.push(ctx.error(`schema of type 'number' with format '${schema.format}' should instead be of type 'integer'`))
-      } else {
-        errors.push(ctx.error(`schema of type 'number' with format '${schema.format}' is invalid, expected one of: ${SCHEMA_NUMBER_FORMATS.join(', ')}`))
+      if (!SCHEMA_NUMBER_FORMATS.includes(schema.format)) {
+        errors.push(ctx.error(`Schema of type 'number' with format '${schema.format}' is invalid. Expected one of: ${SCHEMA_NUMBER_FORMATS.join(', ')}`));
```

Comment:
If we're only allowing non-null valid format then the first `if` on lines 74-77 are redundant because `SCHEMA_NUMBER_FORMATS` doesn't include `null` or `false`, and the error in the first `if` will be triggered in the second `if` anyway.